apps/wallet-mobile/src/features/Swap/useCases/ReviewSwap/TransactionSummary.tsx

```diff
@@ -1,82 +1,65 @@
-    orderData.tokens.priceDenomination,
-    PRICE_PRECISION,
-  )
+  if (tokenInInfo === undefined || tokenOutInfo === undefined) throw new Error('Missing tokenInfos')
+  const amountIn = {
+    info: tokenInInfo,
+    quantity: BigInt((Number(swapForm.createTx?.totalInput) ?? 0) * 10 ** tokenInInfo.decimals),
+  }
+  const amountOut = {
+    info: tokenOutInfo,
+    quantity: BigInt((Number(swapForm.createTx?.totalOutputWithoutSlippage) ?? 0) * 10 ** tokenOutInfo.decimals),
+  }
 
-  const priceImpactRisk = getPriceImpactRisk(Number(priceImpact))
+  const priceImpactRisk = getPriceImpactRisk(Number(swapForm.createTx?.splits[0].priceImpact))
   const priceImpactRiskTheme = usePriceImpactRiskTheme(priceImpactRisk)
-  const priceImpactRiskTextColor = type === 'market' ? priceImpactRiskTheme.text : styles.text.color
-
-  const sellTokenInfo = amounts.sell.info
-  const buyTokenInfo = amounts.buy.info
-
-  const tokenToSellName = sellTokenInfo.ticker ?? sellTokenInfo.name
-  const tokenToBuyName = buyTokenInfo.ticker ?? buyTokenInfo.name
+  const priceImpactRiskTextColor = orderType === 'market' ? priceImpactRiskTheme.text : styles.text.color
 
-  const formattedFeeText = `${Quantities.format(
-    asQuantity((cost.batcherFee.quantity + cost.frontendFeeInfo.fee.quantity).toString()),
-    wallet.portfolioPrimaryTokenInfo.decimals,
-  )} ${wallet.portfolioPrimaryTokenInfo.ticker}`
-  const poolProviderFormatted = capitalize(pool.provider)
-  const poolUrl = getPoolUrlByProvider(pool.provider)
+  const tokenToSellName = tokenInInfo.ticker ?? tokenInInfo.name
+  const tokenToBuyName = tokenOutInfo.ticker ?? tokenOutInfo.name
 
-  const liqFeeQuantity = Quantities.format(asQuantity(cost.liquidityFee.quantity.toString()), sellTokenInfo.decimals)
-  const liqFeeQuantityFormatted = `${liqFeeQuantity} ${tokenToSellName}`
+  const priceInfoValue = `${swapForm.createTx?.netPrice} ${tokenToBuyName}/${tokenToSellName}`
+  const minAdaInfoValue = `${swapForm.createTx?.deposits} ${wallet.portfolioPrimaryTokenInfo.ticker}`
+  const totalFee = `${swapForm.createTx?.totalFee} ${wallet.portfolioPrimaryTokenInfo.ticker}`
+  const minReceivedInfoValue = `${swapForm.createTx?.totalOutput} ${tokenToBuyName}`
 
-  const poolIcon = <PoolIcon providerId={pool.provider} size={18} />
-
-  const priceInfoValue = `${limitDisplayValue} ${tokenToSellName}/${tokenToBuyName}`
-  const minAdaInfoValue = `${Quantities.format(
-    asQuantity(cost.deposit.quantity.toString()),
-    wallet.portfolioPrimaryTokenInfo.decimals,
-  )} ${wallet.portfolioPrimaryTokenInfo.ticker}`
-  const minReceivedInfoValue = `${Quantities.format(
-    asQuantity(calculation.buyAmountWithSlippage.quantity.toString()),
-    buyTokenInfo.decimals,
-  )} ${tokenToBuyName}`
+  const provider = swapForm.createTx?.splits[0]?.dex
 
   const feesInfo = [
+    {
+      label: 'Agreggator',
+      value: <Text style={styles.text}>{swapForm.createTx?.aggregator ?? ''}</Text>,
+      hidden: !isDev,
```

Comment:
- [ ] chore: implement it, pending design

```suggestion
      hidden: !isDev,
```